[Docs] Adding PROMQL functions documentation (#151448)
elastic/elasticsearch: docs/reference/query-languages/promql/promql-limitations.md

- Binary set operators: `and` and `unless`. The `or` operator is supported only at the top level of an expression; nested `or` returns a client error (4xx).
- Group modifiers: `on(...)`, `group_left`, `group_right`
- Functions: see [Not yet supported](functions.md#promql-not-supported) for the full list of recognized but unimplemented functions.

## Native histograms [promql-limitations-native-histograms]

> Prometheus native histograms are not supported yet.

Reviewer: Very likely we have it in 9.5 / GA 

https://github.com/elastic/elasticsearch/issues/150074

cc: @JonasKunz who is working on that.

Reviewer 2: Fine to add this section, I'll update it if we actually get it in.

PromQL functions operate on float samples; series stored as native histograms are not evaluated.

## Metric metadata `help` (HTTP API) [promql-limitations-metadata-help]

On [`/api/v1/metadata`](promql-http-api.md#promql-http-api-metadata-endpoint), each metric includes a `help` string shaped like Prometheus `HELP` lines.

> Currently, the `help` field is always an empty string. Help text from metric definitions is not surfaced yet.

Reviewer: ```suggestion
Metric definition help text is not surfaced yet, so the `help` field remains an empty string.
```
